pkg/appgateserver/metrics.go

```diff
@@ -0,0 +1,36 @@
+package appgateserver
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// relaysTotal is a counter metric that represents the total number of requests processed by the AppGate server.
+	// It is a Counter type metric, which means it is used to accumulate counts of events (in this case, the total number of requests).
+	// This metric is crucial for understanding the overall workload and traffic handled by the server.
+	// It is labeled by 'service_id', which allows for distinguishing and aggregating request counts across different services managed by the AppGate server.
+	//
+	// Example of usage:
+	// - Monitoring the aggregate load and request rate over time.
+	// - Comparing request volumes across different services managed by the AppGate server.
+	relaysTotal metrics.Counter = prometheus.NewCounterFrom(stdprometheus.CounterOpts{
+		Subsystem: "appgateserver",
+		Name:      "requests_total",
+	}, []string{"service_id"})
```

Comment:
What about adding the rpc type here too?

Reply:
Done! Moved where increment is called to support that use case.